controllers/glanceapi_controller.go

```diff
@@ -807,9 +808,14 @@ func (r *GlanceAPIReconciler) reconcileNormal(ctx context.Context, instance *gla
 	}
 	// Mark the Deployment as Ready only if the number of Replicas is equals
 	// to the Deployed instances (ReadyCount), but mark it as True is Replicas
-	// is zero
-	if instance.Status.ReadyCount == *instance.Spec.Replicas || *instance.Spec.Replicas == 0 {
-		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	// is zero. In addition, make sure the controller sees the last Generation
+	// by comparing it with the ObservedGeneration set in the StateFulSet.
+	if (instance.Status.ReadyCount == *instance.Spec.Replicas || *instance.Spec.Replicas == 0) &&
+		(depl.GetStatefulSet().Generation == depl.GetStatefulSet().Status.ObservedGeneration) {
```

Comment:
@gibizer https://github.com/openstack-k8s-operators/lib-common/pull/488 is needed to pass this point in envTests

Reply:
thanks. I landed that lib-common change.

Reply:
Can we be sure that both values we got are the most recent? Operators sdk provides an [api objects cache ](https://logserver.rdoproject.org/70/370/773160e9ba239bb3547715a8e9a406dabf1a1203/github-check/data-plane-adoption-osp-17-to-extracted-crc/ef2f701/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/nova-compute-extraconfig-openstack-openstack-wstzp/logs/openstackansibleee.log) "Using controller-runtime’s split client means that read operations (gets and lists) are read from a cache, and write operations are written directly to the API server."

Reply:
If the status is old (cached) that is OK, we will reconcile as it does not match the generation.
If the status is freshly read bu the spec is old, then we could have a situation where we compare an old spec generation with an old status observed generation and we report readiness too early. However I think it will be not the case here as the createOrUpdate is supposed to return the result of the POST API call instead of doing an independent read after the POST.